events-manager.js

```diff
@@ -201,7 +199,7 @@
 
       _fireCallback(dones, callback, results);
 
-      return values;
+      return results;
```

Comment:
I think we want to return ```_this```. The results are returned in the callback. If we want to populate the results, we use ```.onAsync()``` now the ```results```variable will not be populated here.